js/main.js

```diff
@@ -1,33 +1,50 @@
+"use strict";
+
 /*******************************
  * ASYNC FETCH
-*******************************/
+ *******************************/
 
 // Fetch data and initailize application or handle error
-function appStart () {
-  fetch("https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json")
-  .then((resp) => resp.json())
-  .then((resp) => renderApp(resp))
-  .catch(() => renderErrorMessage());
+function appStart() {
+  fetch(
+    "https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json"
+  )
+    .then((resp) => resp.json())
+    .then((resp) => renderApp(resp))
+    .catch((e) => {
+      console.log(e);
+      renderErrorMessage("appError");
+    });
 }
 
 /*******************************
  * RENDERS
-*******************************/
+ *******************************/
 
 // Render fetched products and add click listeners
 function renderApp(productsList) {
   PRODUCTS_LIST = productsList;
+  CURRENT_PRODUCTS_LIST = [...productsList];
 
   renderProducts();
+  renderBasket();
+
   addProductListeners();
   addLayoutChangeListeners();
+  addsortCurrentListener();
+
+  addSearchListeners();
+  addMobileBasketListeners();
 }
 
 // Render products
 function renderProducts() {
-  PRODUCTS_LIST.forEach((element) => {
-    const list = document.querySelector(".products__list");
+  const list = document.querySelector(".products__list");
 
+  list.innerHTML = "";
+  sortCurrentList();
```

Comment:
funkcja `sortCurrentList` powinna być wywoływana na select `change` event.

Reply:
sortCurrentList jest wywolywana przed każdym renderem produktów dlatego jest czescia funkcji renderProduct. Tak samo jest z eventem change - podczas zmiany musi przerenderowac sie lista ale musi byc to lista posortowana. Dlatego zostalo to zamkniete w jednej funkcji.

Reply:
IMO, ` sortCurrentList()` nie jest tutaj potrzebny. Funkcja `renderProducts()` powinna tylko renderować.
Jak usuniesz tutaj tę funkcję, podepniesz `addSortCurrentListener` do  `sortCurrentList`, a w `sortCurrentList` po sortowaniu wywołasz dodatkowo `renderProducts` w każdym z `case`, to będzie działać poprawnie )

Reply:
Zmienione